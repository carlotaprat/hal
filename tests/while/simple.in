while a + b == c:
    print a, b

while true: pass
