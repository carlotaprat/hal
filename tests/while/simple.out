digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="WHILE_STMT"];
  n1 [label="WHILE_STMT"];
  n2 [label="=="];
  n2 [label="=="];
  n3 [label="+"];
  n3 [label="+"];
  n4 [label="FUNCALL"];
  n4 [label="FUNCALL"];
  n5 [label="a"];
  n6 [label="ARGS"];
  n7 [label="FUNCALL"];
  n7 [label="FUNCALL"];
  n8 [label="b"];
  n9 [label="ARGS"];
  n10 [label="FUNCALL"];
  n10 [label="FUNCALL"];
  n11 [label="c"];
  n12 [label="ARGS"];
  n13 [label="BLOCK"];
  n13 [label="BLOCK"];
  n14 [label="FUNCALL"];
  n14 [label="FUNCALL"];
  n15 [label="print"];
  n16 [label="ARGS"];
  n16 [label="ARGS"];
  n17 [label="FUNCALL"];
  n17 [label="FUNCALL"];
  n18 [label="a"];
  n19 [label="ARGS"];
  n20 [label="FUNCALL"];
  n20 [label="FUNCALL"];
  n21 [label="b"];
  n22 [label="ARGS"];
  n23 [label="WHILE_STMT"];
  n23 [label="WHILE_STMT"];
  n24 [label="true"];
  n25 [label="BLOCK"];
  n25 [label="BLOCK"];
  n26 [label="FUNCALL"];
  n26 [label="FUNCALL"];
  n27 [label="pass"];
  n28 [label="ARGS"];

  n0 -> n1 // "BLOCK" -> "WHILE_STMT"
  n1 -> n2 // "WHILE_STMT" -> "=="
  n2 -> n3 // "==" -> "+"
  n3 -> n4 // "+" -> "FUNCALL"
  n4 -> n5 // "FUNCALL" -> "a"
  n4 -> n6 // "FUNCALL" -> "ARGS"
  n3 -> n7 // "+" -> "FUNCALL"
  n7 -> n8 // "FUNCALL" -> "b"
  n7 -> n9 // "FUNCALL" -> "ARGS"
  n2 -> n10 // "==" -> "FUNCALL"
  n10 -> n11 // "FUNCALL" -> "c"
  n10 -> n12 // "FUNCALL" -> "ARGS"
  n1 -> n13 // "WHILE_STMT" -> "BLOCK"
  n13 -> n14 // "BLOCK" -> "FUNCALL"
  n14 -> n15 // "FUNCALL" -> "print"
  n14 -> n16 // "FUNCALL" -> "ARGS"
  n16 -> n17 // "ARGS" -> "FUNCALL"
  n17 -> n18 // "FUNCALL" -> "a"
  n17 -> n19 // "FUNCALL" -> "ARGS"
  n16 -> n20 // "ARGS" -> "FUNCALL"
  n20 -> n21 // "FUNCALL" -> "b"
  n20 -> n22 // "FUNCALL" -> "ARGS"
  n0 -> n23 // "BLOCK" -> "WHILE_STMT"
  n23 -> n24 // "WHILE_STMT" -> "true"
  n23 -> n25 // "WHILE_STMT" -> "BLOCK"
  n25 -> n26 // "BLOCK" -> "FUNCALL"
  n26 -> n27 // "FUNCALL" -> "pass"
  n26 -> n28 // "FUNCALL" -> "ARGS"

}

